src/mkdocstrings/handlers/python.py

```diff
@@ -290,3 +294,30 @@ def rebuild_category_lists(obj: dict) -> None:
     obj["children"] = [child for _, child in obj["children"].items()]
     for child in obj["children"]:
         rebuild_category_lists(child)
+
+
+def sort_object(obj: CollectorItem, sort_style: str) -> None:
+    """Sort the collected object's children.
+
+    Sorts the object's children list, then each category separately, and then recurses into each.
+
+    Arguments:
+        obj: The collected object, as a dict. Note that this argument is mutated.
+        sort_style: How to sort the children lists - 'lexical' or 'source'.
+    """
+
+    def sort_key(a: dict) -> Any:
+        if sort_style == "lexical":
+            return a.get("name")
```

Comment:
This can break if something is actually missing a name. Then inside `sort` you'll get

```
TypeError: '<' not supported between instances of 'NoneType' and 'str'
```

Maybe this case needs to be handled better, or maybe just use `a["name"]` to get the more explicit error ... or could it be `a.get("name", "")` analogously to your next line?

Although I'd probably prefer non-sortable objects to appear at the end rather than at the beginning.

Reply:
I have a solution to this in 99e7dc4, curious what you think!